src/main/java/com/amazon/ion/bytecode/bin11/ByteArrayBytecodeGenerator11.kt

```diff
@@ -2,14 +2,97 @@
 // SPDX-License-Identifier: Apache-2.0
 package com.amazon.ion.bytecode.bin11
 
+import com.amazon.ion.Decimal
+import com.amazon.ion.IonException
+import com.amazon.ion.Timestamp
+import com.amazon.ion.bytecode.BytecodeGenerator
+import com.amazon.ion.bytecode.bin11.bytearray.OpcodeHandlerTable
+import com.amazon.ion.bytecode.bin11.bytearray.PrimitiveDecoder.readFixedIntAsBigInteger
+import com.amazon.ion.bytecode.bin11.bytearray.TimestampDecoder
+import com.amazon.ion.bytecode.ir.Instructions
+import com.amazon.ion.bytecode.util.AppendableConstantPoolView
+import com.amazon.ion.bytecode.util.ByteSlice
+import com.amazon.ion.bytecode.util.BytecodeBuffer
+import com.amazon.ion.bytecode.util.unsignedToInt
+import com.amazon.ion.impl.bin.utf8.Utf8StringDecoder
+import com.amazon.ion.impl.bin.utf8.Utf8StringDecoderPool
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
+import java.math.BigInteger
+import java.nio.ByteBuffer
 
 @SuppressFBWarnings("EI_EXPOSE_REP2", justification = "constructor does not make a defensive copy of source as a performance optimization")
-internal class ByteArrayBytecodeGenerator11
-@SuppressFBWarnings("URF_UNREAD_FIELD", justification = "field will be read once this class is implemented")
-constructor(
+internal class ByteArrayBytecodeGenerator11(
     private val source: ByteArray,
-    private var i: Int,
-) {
-    // TODO: This should implement BytecodeGenerator
+    private var currentPosition: Int,
+) : BytecodeGenerator {
+    private val utf8Decoder: Utf8StringDecoder = Utf8StringDecoderPool.getInstance().orCreate
+
+    override fun refill(
+        destination: BytecodeBuffer,
+        constantPool: AppendableConstantPoolView,
+        macroSrc: IntArray,
+        macroIndices: IntArray,
+        symTab: Array<String?>
+    ) {
+        var opcode = 0
```

Comment:
0 is a valid opcode. It's probably a little clearer/safer to initialize it to something that is not a valid opcode.
```suggestion
        var opcode = -1
```

Reply:
Good idea